### PR TITLE
Mayh 9612 update topic

### PR DESCRIPTION
**Pull request by comozo**
@lenfree Do you why my update is not working.

### CODE REVIEW

File: kafkaAdmin/resource_topic_create_test.go [hunk 1]
```diff
@@ -97,6 +97,54 @@ resource "kafka_topic" "foobar" {
...
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTopicExists("kafka_topic.foo"),
+					resource.TestCheckResourceAttr(
+						"kafka_topic.foo", "name", "mytopic"),
+					resource.TestCheckResourceAttr(
+						"kafka_topic.foo", "partitions", "2"),
+					resource.TestCheckResourceAttr(
+						"kafka_topic.foo", "replication_factor", "3"),
+					resource.TestCheckResourceAttr(
+						"kafka_topic.foo", "retention_ms", "-1"),
```
Comment (lenfree) [inline kafkaAdmin/resource_topic_create_test.go]: @comozo is this a valid value?

File: kafkaAdmin/resource_topic_create_test.go [hunk 2]
```diff
@@ -26,14 +26,16 @@ func TestAccKafkaAdminTopicCreate(t *testing.T) {
...
 					resource.TestCheckResourceAttr(
-						"kafka_topic.foo", "retention_ms", "300000"),
+						"kafka_topic.foo", "retention_ms", "-1"),
 					resource.TestCheckResourceAttr(
 						"kafka_topic.foo", "segment_bytes", "1073741824"),
-					*/
+					resource.TestCheckResourceAttr(
+						"kafka_topic.foo", "segment_ms", "604800000"),
+					resource.TestCheckResourceAttr(
+						"kafka_topic.foo", "retention_bytes", "-1"),
```
Comment (lenfree) [inline kafkaAdmin/resource_topic_create_test.go]: we accidentally left out  `min_insync_replicas` property, same with createwithconfig test.

**Comment by lenfree** [comment]
@comozo you'd also need this block for update `https://github.com/packetloop/terraform-provider-kafka/blob/master/kafkaAdmin/resource_topic_create.go#L167`.  We also need to add validation to schema type, you could use `https://github.com/lenfree/terraform-provider-singularity/blob/master/mesos_singularity/resource_request.go#L30` as a reference to avoid passing invalid values.

File: kafkaAdmin/resource_topic_create_test.go (hunk 1 above)
Comment (comozo) [reply kafkaAdmin/resource_topic_create_test.go]: @lenfree do you mean retention_ms should be retention.ms ?

Comment (lenfree) [reply kafkaAdmin/resource_topic_create_test.go]: No, sorry for confusion. Please ignore this. I thought -1 is an invalid value but I was wrong.

File: kafkaAdmin/resource_topic_create_test.go [hunk 3]
```diff
@@ -175,9 +165,10 @@ func TopicExistsHelper(s *terraform.State, client *kafka.Client) error {
 	for _, r := range s.RootModule().Resources {
 		id := r.Primary.ID
 
-		// If topic exist, returns error nil.
-		if _, err := client.GetTopic(id); err != nil {
-			return fmt.Errorf("ERROR TOPIC '%s' DOES NOT EXIST: %v", id, err)
+		_, err := client.GetTopic(id)
+		status, _ := errorHelper(err)
+		if status.state == Exists {
```
Comment (lenfree) [inline kafkaAdmin/resource_topic_create_test.go]: @comozo i think you mean 

```go
if status.state == DoesNotExists  {
  			return fmt.Errorf("ERROR TOPIC '%s': %v", id, err)
}
```
?

**Pull request merged**